target_chains/near/receiver/tests/workspaces.rs

```diff
@@ -940,3 +967,164 @@ async fn test_governance_target_fails_if_not_near() {
+            publish_time:      value,
+            prev_publish_time: value,
+            ema_price:         value,
+            ema_conf:          value as u64,
+        };
+        Message::PriceFeedMessage(msg)
+    }
+
+    fn create_accumulator_message_from_updates(
+        price_updates: Vec<MerklePriceUpdate>,
+        tree: MerkleTree<Keccak160>,
+        corrupt_wormhole_message: bool,
+        emitter_address: [u8; 32],
+        emitter_chain: u16,
+    ) -> Vec<u8> {
+        let mut root_hash = [0u8; 20];
+        root_hash.copy_from_slice(&to_vec::<_, BigEndian>(&tree.root).unwrap()[..20]);
+        let wormhole_message = WormholeMessage::new(WormholePayload::Merkle(WormholeMerkleRoot {
+            slot:      0,
+            ring_size: 0,
+            root:      root_hash,
+        }));
+
+        let vaa = wormhole::Vaa {
+            emitter_chain: emitter_chain.into(),
+            emitter_address: wormhole::Address(emitter_address),
+            sequence: 2,
+            payload: (),
+            ..Default::default()
+        };
+
+        let vaa = {
+            let mut cur = Cursor::new(Vec::new());
+            serde_wormhole::to_writer(&mut cur, &vaa).expect("Failed to serialize VAA");
+            cur.write_all(&to_vec::<_, BigEndian>(&wormhole_message).unwrap())
+                .expect("Failed to write Payload");
+            cur.into_inner()
+        };
+
+        let accumulator_update_data = AccumulatorUpdateData::new(Proof::WormholeMerkle {
+            vaa:     PrefixedVec::from(vaa),
+            updates: price_updates,
+        });
+
+        to_vec::<_, BigEndian>(&accumulator_update_data).unwrap()
+    }
+
+    fn create_accumulator_message(
+        all_feeds: &[Message],
+        updates: &[Message],
+        corrupt_wormhole_message: bool,
+    ) -> Vec<u8> {
+        let all_feeds_bytes: Vec<_> = all_feeds
+            .iter()
+            .map(|f| to_vec::<_, BigEndian>(f).unwrap())
+            .collect();
+        let all_feeds_bytes_refs: Vec<_> = all_feeds_bytes.iter().map(|f| f.as_ref()).collect();
+        let tree = MerkleTree::<Keccak160>::new(all_feeds_bytes_refs.as_slice()).unwrap();
+        let mut price_updates: Vec<MerklePriceUpdate> = vec![];
+        for update in updates {
+            let proof = tree
+                .prove(&to_vec::<_, BigEndian>(update).unwrap())
+                .unwrap();
+            price_updates.push(MerklePriceUpdate {
+                message: PrefixedVec::from(to_vec::<_, BigEndian>(update).unwrap()),
+                proof,
+            });
+        }
+        create_accumulator_message_from_updates(
+            price_updates,
+            tree,
+            corrupt_wormhole_message,
+            [1; 32],
+            wormhole::Chain::Any.into(),
+        )
+    }
+
+    let (_, contract, _) = initialize_chain().await;
+
+    // Submit a new Source to the contract, this will trigger a cross-contract call to wormhole
```

Comment:
why do you need to do this for the test? (If you can generate the mock WH governance message, seems like you should be able to mock the price feed update messages as well?)

Reply:
Might not fully understand what you mean but the Wormhole mock is at the contract level, I.E we've mocked out the _whole_ contract, It's not easy to mock out individual functions here without having conditional compilation around contract methods, so for now this is testing a full/real contract.